panic when deleting a namespace in the graph

### Describe the bug

Deleting some namespaces caused a panic: 

panic: namespaces "travel-agency" not found

```
goroutine 29605 [running]:
github.com/kiali/kiali/graph.CheckError({0x54b1860, 0xc000fb4e60})
	/kiali/graph/util.go:52 +0x53
github.com/kiali/kiali/graph/telemetry/istio/appender.getWorkloadList({0xc0015d18e0, 0xa}, {0xc0015d1910, 0xd}, 0xc00208a580)
	kiali/graph/telemetry/istio/appender/appender.go:403 +0x37b
github.com/kiali/kiali/graph/telemetry/istio/appender.getAppWorkloads({0xc0015d18e0, 0xa}, {0xc0015d1910, 0xd}, {0xc0015d18d0, 0x7}, {0xc0015d18ca, 0x2}, 0xc00208a580)
	kiali/graph/telemetry/istio/appender/appender.go:427 +0xdf
github.com/kiali/kiali/graph/telemetry/istio/appender.(*MeshCheckAppender).applyMeshChecks(0xc000dd84b8, {0x54eb0b8, 0xc00310d2c0}, 0xc002e04c30, 0xc00208a580, 0xc002b08828)
	kiali/graph/telemetry/istio/appender/mesh_check.go:79 +0x44c
github.com/kiali/kiali/graph/telemetry/istio/appender.MeshCheckAppender.AppendGraph({0xc002c68420}, {0x54eb0b8, 0xc00310d2c0}, 0xc002e04c30, 0xc00208a580, 0xc002b08828)
	/kiali/graph/telemetry/istio/appender/mesh_check.go:37 +0x65
github.com/kiali/kiali/graph/telemetry/istio.BuildNamespacesTrafficMap({_, _}, {0xc002c68420, {0x0, {0xc000712930, 0x7, 0x7}}, 0x0, 0x1, 0xc002c682d0, ...}, ...)
	kiali/graph/telemetry/istio/istio.go:90 +0x115a
github.com/kiali/kiali/graph/api.graphNamespacesIstio({_, _}, _, {_, _}, {{0x5068fbc, 0x6}, {0x5067a96, 0x5}, {{0xc0008ce974, ...}, ...}, ...})
	/kiali/graph/api/api.go:85 +0x149
github.com/kiali/kiali/graph/api.GraphNamespaces({_, _}, _, {_, _}, {{0x5068fbc, 0x6}, {0x5067a96, 0x5}, {{0xc0008ce974, ...}, ...}, ...})
	/kiali/graph/api/api.go:65 +0x77e
github.com/kiali/kiali/handlers.createGraphGenerator.func1({_, _}, {{0x5068fbc, 0x6}, {0x5067a96, 0x5}, {{0xc0008ce974, 0x15}, {0x45d964b800, {0xc0008ce932, ...}, ...}}, ...})
	kiali/handlers/graph.go:354 +0xde
github.com/kiali/kiali/graph.(*RefreshJob).refresh(0xc002b2e6c8)
	/kiali/graph/refresh_job.go:274 +0x67f
created by github.com/kiali/kiali/graph.(*RefreshJob).Start in goroutine 23990
	/kiali/graph/refresh_job.go:106 +0xbaf
```

### Expected Behavior
In case of deleting a namespace, should throw an error, but avoid a panic

### What are the steps to reproduce this bug?

- Go to the traffic graph
- See, for example, the bookinfo namespace graph
- Delete bookinfo namespace

1. …
2. …
3. …

### Environment
*Learn about how to determine versions [here](https://kiali.io/docs/faq/general/#how-do-i-determine-what-version-i-am-running).*

- **Kiali version:** master
- **Istio version:**
- **Kubernetes impl:** 
- **Kubernetes version:** 
- **Other notable environmental factors:**

--- a/graph/refresh_job.go
+++ b/graph/refresh_job.go
@@ -2,6 +2,7 @@ package graph
 
 import (
 	"context"
+	"runtime/debug"
 	"sync"
 	"time"
 
@@ -234,6 +235,15 @@ func (j *RefreshJob) cleanup() {
 // 4. Generates a fresh graph
 // 5. Updates the cache
 func (j *RefreshJob) refresh() {
+	defer func() {
+		if recovered := recover(); recovered != nil {
+			log.Errorf("Panic while refreshing graph cache for session [%s]: %v\n%s", j.sessionID, recovered, string(debug.Stack()))
+			// A panic indicates a broken refresh cycle. Evict the stale graph and stop the job.
+			j.cache.Evict(j.sessionID)
+			j.Stop()
+		}
+	}()
+
 	// Check if session's graph still exists (use internal method to not update LastAccessed)
 	cached, found := j.cache.getSessionGraphInternal(j.sessionID)
 	if !found {

--- a/graph/refresh_job_test.go
+++ b/graph/refresh_job_test.go
@@ -270,6 +270,51 @@ func TestRefreshJob_GeneratorError(t *testing.T) {
 	assert.Equal(t, len(trafficMap), len(retrieved.TrafficMap))
 }
 
+func TestRefreshJob_GeneratorPanic(t *testing.T) {
+	ctx := context.Background()
+	config := &GraphCacheConfig{
+		Enabled:           true,
+		InactivityTimeout: 5 * time.Minute,
+		MaxCacheMemoryMB:  50,
+		RefreshInterval:   1 * time.Hour,
+	}
+	cache := NewGraphCache(ctx, config).(*GraphCacheImpl)
+
+	generator := func(ctx context.Context, options Options) (TrafficMap, error) {
+		panic("boom")
+	}
+	cache.SetGraphGenerator(generator)
+
+	options := Options{
+		TelemetryOptions: TelemetryOptions{
+			CommonOptions: CommonOptions{
+				Duration: 120 * time.Second,
+			},
+		},
+	}
+
+	// Set initial cached graph
+	trafficMap := createTestTrafficMap(5)
+	cached := &CachedGraph{
+		LastAccessed:    time.Now(),
+		Options:         options,
+		RefreshInterval: 30 * time.Second,
+		Timestamp:       time.Now(),
+		TrafficMap:      trafficMap,
+	}
+	err := cache.SetSessionGraph("test-session", cached)
+	require.NoError(t, err)
+
+	job := NewRefreshJob(ctx, "test-session", options, cache, generator, 30*time.Second)
+
+	// refresh should recover from panic and not crash the test
+	job.refresh()
+
+	assert.True(t, job.stopped)
+	_, found := cache.GetSessionGraph("test-session")
+	assert.False(t, found)
+}
+
 func TestRefreshJob_StartAndStop(t *testing.T) {
 	ctx := context.Background()
 	config := &GraphCacheConfig{

